modules/caddyhttp/fileserver/staticfiles.go

```diff
@@ -95,6 +96,13 @@ func (fsrv *FileServer) ServeHTTP(w http.ResponseWriter, r *http.Request, _ cadd
 	root := repl.ReplaceAll(fsrv.Root, ".")
 	suffix := repl.ReplaceAll(r.URL.Path, "")
 	filename := sanitizedPathJoin(root, suffix)
+	contentTypes := r.URL.Query()["downloadAs"]
```

Comment:
Let's rename it, also use Get:

```go
contentType := r.URL.Query().Get("archive")
```

If the archive type isn't supported, return a HTTP 406 (Not Acceptable). No fallbacks.

Reply:
I used here [1] 415 unsupported media type, don't you think it is more appropriate?
[1] https://github.com/jaysonsantos/caddy/blob/660f8deb2e240741ff998286057252faba3e3cfd/modules/caddyhttp/fileserver/browse.go#L272-L275

Reply:
No, because [415 means](https://httpstatuses.com/415):

> The origin server is refusing to service the request because the payload is in a format not supported by this method on the target resource.

Which doesn't apply here. [406 is precisely what we want](https://httpstatuses.com/406):

> The target resource does not have a current representation that would be acceptable to the user agent, according to the proactive negotiation header fields received in the request, and the server is unwilling to supply a default representation.